torch/_dynamo/allowed_functions.py

```diff
@@ -309,7 +309,40 @@ def _builtin_constant_ids():
     return rv
 
 
+# A subcheck of is_allowed, we utilize this for patching is_allowed around distributed.
```

Comment:
@yanboliang curious how you would rewrite this code after your refactor

Reply:
After reading the comments here, it seems we are trying to inline these functions rather than put them as FX graph node. After my refactor, these functions should __NOT in the allow function list__ and should __in the force inline list__. I think it should be easy to add them to the right list if my refactor really works well.
The principle of my refactor is to make every function __ONLY__ fall into one of the three: allow_in_graph, inline, skip.